handout.md

```diff
@@ -17,6 +18,57 @@ Start git at a folder
 git init
 ```
 
+Checkout a branch
+```
+git checkout <branch_name>
+```
+
+Creating a new branch
+```
+git branch -b <new_branch_name>
+```
+
+Show all branches
+```
+git branch -a
+```
+
+Commit changes to local repository
+```
+git commit -m"commit_message"
+```
+
 ### Working with Remote Repos
+First we need to add a remote location.
+
+```
+git remote add <name> <url>
+```
+
+
+## Advanced topics
+### Ignoring files in your repository
+To ignore specific files, you need to create a .gitignore by running
+```
+touch .gitignore
+```
+
+### Cleaning generated files
+Removing files that were recently created that you do not want to ever commit.
+This is done in two steps:
+```
+git add .
+git reset --hard HEAD
+```
+
+### Removing unwanted directories permanently
+
+```
+git filter-branch --tree-filter 'rm -rf lib' HEAD
+```
+
+This is is if you want to completely remove any trace of the diretory `lib` from your repository.
+
```

Comment:
I'd recommend adding in the following to this:
This command modifies every commit in the current branch, back to the root, and changes it by removing the lib folder.  As such, it will modify the commitish for all of those commits, and hence likely change public history.  This is a good command when preparing a migrated repository, but a bad idea in most other cases.